src/pages/workspaces/workspace/Dashboard/DashboardAuthContainer.tsx

```diff
@@ -0,0 +1,44 @@
+import { cond, DEFAULT } from '@terra-ui-packages/core-utils';
+import _ from 'lodash/fp';
+import React, { ReactNode, useEffect, useState } from 'react';
+import { centeredSpinner } from 'src/components/icons';
+import { Ajax } from 'src/libs/ajax';
+import { useStore } from 'src/libs/react-utils';
+import { authStore } from 'src/libs/state';
+import SignIn from 'src/pages/SignIn';
+import { WorkspaceDashboardPage } from 'src/pages/workspaces/workspace/Dashboard/WorkspaceDashboardPage';
+import DashboardPublic from 'src/pages/workspaces/workspace/DashboardPublic';
+
+export interface DashboardAuthContainerProps {
+  namespace: string;
+  name: string;
+}
+
+export const DashboardAuthContainer = (props: DashboardAuthContainerProps): ReactNode => {
```

Comment:
Why is this a TSX file instead of a plain ts file?

Reply:
1. While it's low priority, we're moving to TSX as the preferred format generally
2. It made it easier to work with copilot, since TSX is the more standard way of doing things